src/message.rs

```diff
@@ -21,6 +22,10 @@ fn from_cbor_blake2b256<S: serde::ser::Serialize>(obj: &S) -> Result<Cid, Error>
     ))
 }
 
+pub fn parse_address(s: &str) -> anyhow::Result<Address> {
+    Ok(Address::from_str(s).or_else(|_| Address::new_delegated(4, s.as_ref()))?)
```

Comment:
This doesn't quite work. Mainnet and Testnet addresses have to be parsed separately, and `0x` addresses have to be parsed as hex before being passed to `Address::new_delegated`. I'll fix these issues.

Reply:
Got it.

Hubert explained some of the intricacies of the parsing. Given this is just an input to a JSON RPC call, does it make sense to implement basic sanity checks (prefix, length) and avoid using the Address type?

Reply:
IMO, we should **not** avoid it:
1. We get better control over errors. In the case of sending a potentially bogus input to the RPC node, we don't know how it will behave and what kind of error it will output. Also, we get that error later (if any), impacting _the user_ experience.
2. We may get away with the problem for this particular use case, but the next person implementing something with the addresses might stumble into this. Then we'll pay the cost of refactoring (or more workarounds).
3. Arguably, the amount of work is similar. If you write your own checkers, you better provide some tests for them. In the case of re-using the `Address` type, we already assume it's well-tested. The _smell_ comes from having to copy-paste some `to_filecoin_address` (from Ethereum address) logic from Forest, as it's not easily importable. As you mentioned, it might make sense to start creating some `forest-types` crate, if not now, then as soon as the amount of copypasta becomes unbearable.
4. It's a good exercise. We begin knowing which interfaces we should make more accessible from Forest.

Does it make sense?

Reply:
I've updated the method based on feedback. I also added a more thorough test